java-frontend/src/test/java/org/sonar/java/se/xproc/ExceptionalCheckBasedYieldTest.java

```diff
@@ -0,0 +1,182 @@
+    assertThat(seCheckExceptionalYield.exceptionType().isSubtypeOf("java.lang.RuntimeException")).isTrue();
+  }
+
+  private static class TestSECheck extends SECheck {
+
+    private static final MethodMatcher MATCHER = MethodMatcher.create().typeDefinition("foo.bar.A").name("plantFlowers").addParameter("boolean");
+
+    @Override
+    public ProgramState checkPreStatement(CheckerContext context, Tree syntaxNode) {
+      ProgramState state = context.getState();
+      if (syntaxNode.is(Tree.Kind.METHOD_INVOCATION) && MATCHER.matches((MethodInvocationTree) syntaxNode)) {
+        SymbolicValue param = state.peekValue();
+        Constraint paramConstraint = state.getConstraint(param);
+        if (paramConstraint instanceof BooleanConstraint && ((BooleanConstraint) paramConstraint) == BooleanConstraint.TRUE) {
+          // create new yield with the exception
+          context.addExceptionalYield(param, state, "java.lang.UnsupportedOperationException", this);
+          // interrupt exploration (theoretical runtime exception)
+          return null;
+        }
+      }
+      return state;
+    }
+  }
+
+  @Test
+  public void exceptionType_is_required() {
+    thrown.expect(IllegalArgumentException.class);
+    final Class<? extends SECheck> seCheckClass = new SECheck() {
+    }.getClass();
+    Type exceptionType = null;
+    new ExceptionalCheckBasedYield(exceptionType, seCheckClass, null, mockMethodBehavior());
+  }
+
+  @Test
+  public void exceptionType_cannot_be_changed() {
+    thrown.expect(UnsupportedOperationException.class);
+    final Class<? extends SECheck> seCheckClass = new SECheck() {
+    }.getClass();
+    Type exceptionType = mock(Type.class);
+    ExceptionalCheckBasedYield yield = new ExceptionalCheckBasedYield(exceptionType, seCheckClass, null, mockMethodBehavior());
+    yield.setExceptionType(mock(Type.class));
+  }
+
+  @Test
+  public void test_toString() {
+    Type exceptionType = mock(Type.class);
+    when(exceptionType.fullyQualifiedName()).thenReturn("org.foo.MyException");
+    ExceptionalCheckBasedYield yield = new ExceptionalCheckBasedYield(exceptionType, SECheck.class, null, mockMethodBehavior());
+
+    assertThat(yield.toString()).isEqualTo("{params: [], exceptional (org.foo.MyException), check: SECheck}");
+  }
+
+  @Test
+  public void test_equals() {
+    final Class<? extends SECheck> seCheckClass1 = new SECheck() {
+    }.getClass();
+    final Class<? extends SECheck> seCheckClass2 = (new SECheck() {
+    }).getClass();
+
+    MethodBehavior mb = mockMethodBehavior();
+
+    Type mockedExceptionType1 = mock(Type.class);
+
+    ExceptionalCheckBasedYield yield = new ExceptionalCheckBasedYield(mockedExceptionType1, seCheckClass1, null, mb);
+    ExceptionalYield otherYield = new ExceptionalCheckBasedYield(mockedExceptionType1, seCheckClass1, null, mb);
+
+    assertThat(yield).isNotEqualTo(null);
```

Comment:
disagree, tests contracts of equals method... condition coverage purpose. You prefer `assertThat(yields.equals(null)).isFalse();` ?